core/src/main/java/hudson/slaves/ChannelPinger.java

```diff
@@ -122,6 +124,14 @@ protected void onDead(Throwable cause) {
                     LOGGER.log(SEVERE,"Failed to terminate the channel "+channel.getName(),e);
                 }
             }
+            /** Keep in a separate method so we do not even try to do class loading on {@link PingFailureAnalyzer} from an agent JVM. */
+            private void analyze(Throwable cause) throws IOException {
+                for (PingFailureAnalyzer pfa : PingFailureAnalyzer.all()) {
+                    pfa.onPingFailure(channel,cause);
```

Comment:
It would be nice, though (a) there are no implementations to test, as noted in the description, (b) I spent some time testing the patch as written and would have to spend more time retesting with any other modifications.